src/tess.py

```diff
@@ -0,0 +1,359 @@
+import os
+import glob
+import requests
+from datetime import datetime
+import numpy as np
+import pandas as pd
+import sqlalchemy as sa
+import warnings
+import socket
+
+from timeit import default_timer as timer
+
+from astroquery.mast import Observations, Catalogs
+from astropy.table import Table
+import astropy.io.fits as fits
+
+from src.source import angle_diff
+from src.observatory import VirtualObservatory, ParsObservatory
+from src.catalog import Catalog
+from src.dataset import DatasetMixin, RawPhotometry, Lightcurve
+
+class ParsObsTESS(ParsObservatory):
+
+    # must register this observatory in list of allowed names
+    ParsObservatory.add_to_obs_names("TESS")
+
+    def __init__(self, **kwargs):
+
+        super().__init__("tess")
+
+        self.distance_thresh = self.add_par(
+            "distance_thresh",
+            10.0, 
+            float, 
+            "Distance threshold in arcseconds for flagging "
+            "close stars while querying TIC."
+        )
+
+        self.magdiff_thresh = self.add_par(
+            "magdiff_thresh",
+            0.75, 
+            float, 
+            "Magnitude difference threshold for flagging "
+            "similar stars while querying TIC."
+        )
+
+        self.cat_qry_radius = self.add_par(
+            "cat_qry_radius",
+            0.01,
+            float,
+            "Radius in degrees for cone search for MAST Catalogs query."
+            "This query is to find the TESS ID for given source."
+        )
+
+        self.obs_qry_radius = self.add_par(
+            "obs_qry_radius",
+            0.1,
+            float,
+            "Radius in degrees for cone search for MAST Observations query."
+            "This query is to find data from TIC for given source."
+        )
```

Comment:
To be honest, I'm not sure either, I made it two different values for more customizability. For Catalogs.query_region, radius is for cone search by given ra+dec, which makes sense. But for Observations.query_criteria, I guess it doesn't actually need a radius because it queries by objectname; however, I'm not sure why but the query returns a list of TESS observations that all don't have any ID or name matching the given objectname. When you add a radius argument to the query, the list gets filtered down by "distance", which is defined as "Angular separation between searched coordinates and center of obsevation" even though I never provide any coordinates to the query. Kind of mysterious. Here is the api: https://astroquery.readthedocs.io/en/latest/api/astroquery.mast.ObservationsClass.html#astroquery.mast.ObservationsClass.query_criteria

Reply:
Maybe this is the distance measured from the TESS catalog source you are providing. 
In any case, giving a much bigger radius may be the reason we are getting too many TICIDs. 
But I think maybe giving just one radius, and being fairly generous (like 10 arcsec) would be good enough. 
And then need to make sure we only return a single TICID (see my latest email).

Reply:
Got it